src/views/RoomDetail.vue

```diff
@@ -49,17 +64,92 @@ export default {
   created () {
     this.roomID = this.$route.params.roomID
   },
-  mounted () {
-    const db = this.$firebase.firestore()
-    db
-      .collection('Rooms')
-      .doc(this.roomID)
-      .get()
-      .then(doc => {
-        this.room = doc.data()
-      })
+  async mounted () {
+    await this.getRoomKeyValuePairs()
+    await this.getRoomCheckIns()
+    await this.getRoomCheckInsTomorrow()
+    await this.fetchUserIDList()
+    await this.isUserSignedIn()
+    await this.isUserSignedInTomorrow()
+    await this.roomCapacityToday()
+    await this.roomCapacityTomorrow()
   },
   methods: {
+    getRoomKeyValuePairs: async function () {
+      await $db()
+        .collection('Rooms')
+        .doc(this.roomID)
+        .get()
+        .then(doc => {
+          this.room = doc.data()
+          this.room.checkIns = []
+          this.room.checkInsTomorrow = []
+        })
+    },
+    getRoomCheckIns: async function () {
+      await $db()
+        .collection('Rooms')
+        .doc(this.roomID)
+        .collection('CheckIn')
+        .where('date', '==', this.currentDate)
+        .get()
+        .then(querySnapshot => {
+          querySnapshot.forEach((checkIn) => {
+            this.room.checkIns = checkIn.data()
+          })
+        })
+    },
+    getRoomCheckInsTomorrow: async function () {
+      await $db()
+        .collection('Rooms')
+        .doc(this.roomID)
+        .collection('CheckIn')
+        .where('date', '==', this.dateTomorrow())
+        .get()
+        .then(querySnapshot => {
+          querySnapshot.forEach((checkIn) => {
+            this.room.checkInsTomorrow = checkIn.data()
+          })
+        })
+    },
+    fetchUserIDList: async function () {
+      if (this.room.checkIns.user) {
+        this.room.checkIns.user.forEach(user => this.userIdList.todayID.push(user.id))
+      }
+      if (this.room.checkInsTomorrow.user) {
+        this.room.checkInsTomorrow.user.forEach(user => this.userIdList.tomorrowID.push(user.id))
+      }
+    },
+    isUserSignedIn: async function () {
```

Comment:
Kann man hier nicht den Capacity check direkt mit einbauen? Die Funktionen machen doch das gleiche, nämlich den Button deaktivieren.
```suggestion
isUserSignedIn: async function () {
    const currentUser = await userService.currentUser()
    const isSignedIn = this.userIdList.tomorrowID.some(userId => userId === currentUser.id)
    const noCapacity = this.room.checkIns.user.length >= this.totalSteps()
    if(isSignedIn || noCapacity) {
    this.disableButtonToday = true
    }
  }
```

Reply:
Richtig, ginge ja. Möchte aber keine Abhängigkeit verschiedener Validierungen in einer Methode haben.

Reply:
Aber die Methoden machen doch genau das selbe. Sie deaktivieren den Button. Dafür sollten wir keine zwei Methoden haben sondern das sollte in einer zusammengefasst sein. Es entstehen ja auch keine Abhängigkeiten dadurch.